File: AquariumObject.cpp
#include "AquariumObject.hpp"

AquariumObject::AquariumObject(Aquarium& _aquarium, float _speed, Vector2 _position) : aquarium(_aquarium), speed(_speed) {
    position = _position;
}

Aquarium& AquariumObject::getAquarium() const {
    return aquarium;
}

float AquariumObject::getSpeed() const {
    return speed;
}

Vector2 AquariumObject::getPosition() const {
    return position;
}

void AquariumObject::setPosition(Vector2 _position) {
    position = _position;
}

void AquariumObject::tick() {
    move();
}

File: production/Java/src/AquariumObject.java
public abstract class AquariumObject {
  private final Aquarium aquarium;
  private final float speed;
  private Vector2 position;

  public AquariumObject(Aquarium aquarium, float speed, Vector2 position) {
    this.aquarium = aquarium;
    this.speed = speed;
    this.position = position;
  }

  public Aquarium getAquarium() {
    return aquarium;
  }

  public float getSpeed() {
    return speed;
  }

  public Vector2 getPosition() {
    return position;
  }

  public void setPosition(Vector2 position) {
    this.position = position;
  }

  public abstract void move();

  public void tick() {
    move();
  }
}